grakn-graph/src/main/java/ai/grakn/graph/internal/computer/GraknComputerImpl.java

```diff
@@ -57,33 +67,41 @@
 
     public GraknComputerImpl(Graph graph) {
         this.graph = graph;
-        if(graph instanceof TinkerGraph){
+        if (graph instanceof TinkerGraph) {
             graphComputerClass = TinkerGraphComputer.class;
         } else {
             graphComputerClass = GraknSparkComputer.class;
         }
     }
 
     @Override
-    public ComputerResult compute(VertexProgram program, MapReduce... mapReduces) {
+    public ComputerResult compute(Boolean includesShortcut, Set<LabelId> types,
+                                  VertexProgram program, MapReduce... mapReduces) {
         try {
-            graphComputer = getGraphComputer().program(program);
+            if (program != null) graphComputer = getGraphComputer().program(program);
```

Comment:
I vote for annotating `program` param as `@Nullable`! Same for methods below as well.